### PR TITLE
Dev - Kendo ComboBox for Categories upon contest creation

### PR DESCRIPTION
**Pull request by jicata**
https://github.com/SoftUni-Internal/suls-issues/issues/2299

### CODE REVIEW

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs [hunk 1]
```diff
@@ -225,7 +225,7 @@ public ActionResult GetLatestContests([DataSourceRequest]DataSourceRequest reque
             return this.PartialView(GlobalConstants.QuickContestsGrid, latestContests);
         }
 
-        public JsonResult GetCategories()
+        public JsonResult GetCategories(string contestFilter)
```
Comment (KristianMariyanov) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs]: Are you sure, that you don't break some other functionality?

File: Open Judge System/Web/OJS.Web/Areas/Administration/Views/Contests/EditorTemplates/CategoryDropDown.cshtml [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+      .Placeholder("Select category")
+      .DataTextField("Name")
+      .DataValueField("Value")
+      .DataSource(data =>
+      {
+          data.ServerFiltering(true)
+              .Read(read => read.Action("GetCategories", "Contests").Data("filterContests"));
+      })
+      .SelectedIndex(0)
+      .HtmlAttributes(new{style="width:100%"}))
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Views/Contests/EditorTemplates/CategoryDropDown.cshtml]: spaces

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs (hunk 1 above)
Comment (jicata) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs]: Yes, the only place where the GetCategories method is called is in the CategoryDropDown.cshtml

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs [hunk 3]
```diff
@@ -237,11 +237,12 @@ public JsonResult GetCategories()
             }
 
             var dropDownData = categories
+                .Where(c => c.Name.ToLower().Contains(contestFilter.ToLower()))
                 .ToList()
                 .Select(cat =>
                     new
                     {
-                        Parent = cat.Parent != null ? cat.Parent.Name : null,
+                        Parent = cat.Parent?.Name,
```
Comment (YanaSlavcheva) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs]: you may use `c`, not `cat`

**Pull request merged**